src/parser.cpp

```diff
@@ -769,11 +782,20 @@ Writer<std::unique_ptr<AST::AST>> Parser::parse_if_statement() {
 		return result;
 	}
 
-	auto e = std::make_unique<AST::IfStatement>();
+	auto e = std::make_unique<AST::IfElseStatement>();
 
 	e->m_condition = std::move(condition.m_result);
 	e->m_body = std::move(body.m_result);
 
+	if (consume(token_type::KEYWORD_ELSE)) {
+		auto else_body = parse_statement();
+
+		if (handle_error(result, else_body))
+			return result;
+
+		e->m_else_body = std::move(else_body.m_result);
+	}
+
```

Comment:
It would be nicer if the parsing of the else body was hoisted to before the creation of the AST node

Reply:
I think doing this requires that I write the whole type when declaring `else_body`, because I have to call `parse_statement` after checking for `KEYWORD_ELSE` so `auto` wouldn't work. Isn't that worse? Here is the changed code:
```c++
        Writer<std::unique_ptr<AST::AST>> else_body; 

        if (consume(token_type::KEYWORD_ELSE)) {
                else_body = parse_statement();

                if (handle_error(result, else_body))
                        return result;
        }   

        auto e = std::make_unique<AST::IfElseStatement>();

        e->m_condition = std::move(condition.m_result);
        e->m_body = std::move(body.m_result);

        if (else_body.m_result)
                e->m_else_body = std::move(else_body.m_result);
```

Reply:
do as you like, the pr is approved nonetheless